components/datadog/agentparams/params.go

```diff
@@ -88,24 +90,34 @@ func WithVersion(version string) func(*Params) error {
 // WithPipelineID uses a specific testing pipeline ID of the datadog-agent CI. For example: `16497585`
 func WithPipelineID(pipelineID string) func(*Params) error {
```

Comment:
To do this, I think I need to modify `CommonEnvironment.AgentPipelineID`, which currently is:
```
func (e *CommonEnvironment) AgentPipelineID() string {
	return e.AgentConfig.Get(DDAgentPipelineIDParamName)
}
```

to something like:
```
func (e *CommonEnvironment) AgentPipelineID() (int, error) {
        // Logic to parse the pipeline id, and return an error if it fails
}
```
because I want to be able to stop the execution when parsing (I don't want to ignore the error, which could lead to surprising results for the user), but:
- Do we allow methods from `CommonEnvironment` to return errors (I haven't seen any that does)?
- Do we allow the framework to fail when given an invalid input, or should it continue while ignoring the error (which is the behavior of the `GetIntWithDefault` method that already exists)?